plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -1092,4 +1093,13 @@ public static long getModificationTime(String path, TrinoFileSystem fileSystem)
             throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed to get file modification time: " + path, e);
         }
     }
+
+    public static Optional<String> getDataLocation(Map<String, String> properties)
+    {
+        if (properties.containsKey(WRITE_DATA_LOCATION) && properties.get(WRITE_DATA_LOCATION) != null) {
+            return Optional.of(properties.get(WRITE_DATA_LOCATION));
+        }
+
+        return Optional.empty();
```

Comment:
Consider simplifying this logic like `Optional.ofNullable(properties.get(WRITE_DATA_LOCATION))`.  No need to extract a method as it's very simple.